unit-tests/py/rspy/devices.py

```diff
@@ -290,33 +290,51 @@ def enabled():
     return { device.serial_number for device in _device_by_sn.values() if device.enabled }
 
 
-def by_product_line( product_line ):
+def by_product_line( product_line, ignored_products ):
     """
     :param product_line: The product line we're interested in, as a string ("L500", etc.)
+    :param ignored_products: List of products we want to ignore. e.g. ['D455', 'D457', etc.]
     :return: A set of device serial-numbers
     """
     global _device_by_sn
-    return { device.serial_number for device in _device_by_sn.values() if device.product_line == product_line }
+    result = set()
+    for device in _device_by_sn.values():
+        ignore_flag = False
+        for ignored_product in ignored_products:
+            if ignored_product in device.name:
+                ignore_flag = True
+                break
+        if device.product_line == product_line and not ignore_flag:
+            result.add(device.serial_number)
+    return result
 
 
-def by_name( name ):
+def by_name( name, ignored_products ):
     """
     :param name: Part of the product name to search for ("L515" would match "Intel RealSense L515")
+    :param ignored_products: List of products we want to ignore. e.g. ['D455', 'D457', etc.]
     :return: A set of device serial-numbers
     """
     global _device_by_sn
-    return { device.serial_number for device in _device_by_sn.values() if device.name  and  device.name.find( name ) >= 0 }
-
+    result = set()
+    ignored_list_as_str = " ".join(ignored_products)
+    for device in _device_by_sn.values():
+        if (name not in ignored_list_as_str) and (device.name.find( name ) >= 0):
```

Comment:
`name not in ignored_list_as_str` is invariant.
Maybe you meant `device.name`? But then: device.name is `RealSense L515` and the ignored list is just `L515`, so it won't work...

Reply:
name is the given parameter for the functions and its "L515" in this case
see the function signature and documentation

Reply:
But it's invariant -- meaning it does not change from one loop iteration to the next... you want to compare the `device.name`, right?

Reply:
This function get two parameters (name, ignored_prodcuts)
lets say NAME="D450" and ignored_products = ['D457', 'L515']
now, I want to loop over the devices, in order to get the serial number of the "D450" device
to do that, I go over devices, asking if the name(D450) not in the ignored products, and if the device.name I'm looping on contains the "D450" string.
So, I still need "name" in the if, since its is the input I want to make sure its not in the ignored products
The device.name can work somehow, but it gives me a long string ("D450 Intel RealSense..."), that I need to extract the "D450" from it, and then check if its doesn't exist in the ignored_list_as_str